tests/components/accuweather/__init__.py

```diff
@@ -1 +1,44 @@
 """Tests for AccuWeather."""
+import json
+
+from homeassistant.components.accuweather.const import DOMAIN
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry, load_fixture
+
+
+async def init_integration(
+    hass, forecast=False, unsupported_icon=False
+) -> MockConfigEntry:
+    """Set up the AccuWeather integration in Home Assistant."""
+    options = {}
+    if forecast:
+        options["forecast"] = True
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        title="Home",
+        unique_id="0123456",
+        data={
+            "api_key": "32-character-string-1234567890qw",
+            "latitude": 55.55,
+            "longitude": 122.12,
+            "name": "Home",
+        },
+        options=options,
+    )
+
+    current = json.loads(load_fixture("accuweather/current_conditions_data.json"))
+    forecast = json.loads(load_fixture("accuweather/forecast_data.json"))
+
+    if unsupported_icon:
+        current["WeatherIcon"] = 999
+
+    with patch(
+        "accuweather.AccuWeather.async_get_current_conditions", return_value=current,
+    ), patch("accuweather.AccuWeather.async_get_forecast", return_value=forecast):
```

Comment:
I believe you need to use AsyncMock here
`from tests.async_mock import AsyncMock`

Reply:
You may need to revert your patching location to be back inside homeassistant see https://docs.python.org/3/library/unittest.mock.html#where-to-patch.

Reply:
I can't figure out how to suggest the exact changes directly, but I hope it helps

Reply:
`patch` will patch target coroutines with AsyncMock by default.

Reply:
We can patch the library but use the path in home assistant.

Reply:
You are right, only the coroutine method is patched here.

Reply:
i was just trying to pull straws to resolve the failure and notice it was different from the rest so was seeing if it was some conflict with patching same thing in two different ways. ideally they should all use the homeassistant "path" though